File: CPP-Python-ROS-Apps/python/DemoApp2/demoapp2.py
#!/usr/bin/env python
import time
import argparse
from flyt_python import api

drone = api.navigation(timeout=120000)  # instance of flyt droneigation class

# at least 3sec sleep time for the drone interface to initialize properly
time.sleep(3)

## parsing command line arguments
parser = argparse.ArgumentParser(description='Process a float value.')
parser.add_argument('side', metavar='side_length', type=float, help='side length of the square')
args = parser.parse_args()

## lets fly 
side_length = args.side

print "taking off!"
drone.take_off(5.0)

print 'flying in square', side_length
drone.position_set(side_length, 0, 0, relative=True)
drone.position_set(0, side_length, 0, relative=True)
drone.position_set(-side_length, 0, 0, relative=True)
drone.position_set(0, -side_length, 0, relative=True)

print "landing"
drone.land(False)
print 'Cheers!!'

# shutdown the instance
drone.disconnect()

File: CPP-Python-ROS-Apps/cpp/DemoApp2/demoapp2.cpp
#include <cpp_api/navigation_bridge.h>
#include <iostream>
#include <ros/package.h>

Navigation nav;
int main(int argc, char *argv[])
{
	if(argc < 2){
		std::cout<<"\nThis app expects arguments\n";
		exit(0);
	}
	
    float side_length = std::stof(argv[1]);             //Convert Argument from string to float
    nav.take_off(3.0);                                   //Taking Off
    nav.position_set(side_length,0,-3);                 //Sending Position Setpoints with side length accepted from script
    nav.position_set(side_length,side_length,-3);
    nav.position_set(0,side_length,-3);
    nav.position_set(0,0,-3);
    nav.land(false);                                         //Landing
}
